R/utils.R

```diff
@@ -288,3 +288,35 @@ flname_xos_get <- function(fl, package = "dpbuild") {
   flname <- basename(fl_path)
   return(flname)
 }
+
+
+#' @title Check pins package compatibility
+#' @description Check pins package compatibility
+#' @param is_legacy A boolean variable to indicate if the pins version is legacy one
+#' @keywords internal
+check_pins_compatibility <- function(){
+  read_conf_file <- dpbuild:::dpconf_read(project_path = ".")
+
+  is_legacy_key_in_config <- "is_legacy" %in% names(read_conf_file)
+  is_legacy_pins_in_config <- as.logical(read_conf_file$is_legacy)
```

Comment:
For simplicity, I think you should combine these two to one single option: "is_legacy_dp", which would only be true if both of these conditions are met. Note that if the key doesn't exist, the second boolean here will actually be NA, not FALSE, so it won't behave as expected in the all statement below. Also, I think the as.logical is unnecessary here